src/platform_impl/linux/wayland/window.rs

```diff
@@ -512,26 +523,25 @@ impl WindowStore {
         F: FnMut(WindowStoreForEach<'_>),
     {
         for window in &mut self.windows {
+            let prev_scale_factor = window.current_scale_factor;
+            if let Some(scale_factor) = window.new_scale_factor {
+                window.current_scale_factor = scale_factor;
```

Comment:
I'm not really a fan that you're updating scale factor right here, but sending new event in event loop. However I feel like it's already complex to handle such things correctly in a callback, so add a comment at least explain that scale factor change is the first event.

Reply:
I just moved it from after to before the callback.
I don't know what is the split between window/event_loop for and the criteria for what should be on either side ?

Reply:
The goal is to not go out of sync with the library user, right now it's fine, and knowing that backend will be changed anyway let's just keep it.